// appstoreconnect/dist/v1/routes/provisioning/bundle-id-capabilities/types.js
'use strict'
Object.defineProperty(exports, '__esModule', { value: true })
var CapabilityType
;(function(CapabilityType) {
    CapabilityType['ICLOUD'] = 'ICLOUD'
    CapabilityType['IN_APP_PURCHASE'] = 'IN_APP_PURCHASE'
    CapabilityType['GAME_CENTER'] = 'GAME_CENTER'
    CapabilityType['PUSH_NOTIFICATIONS'] = 'PUSH_NOTIFICATIONS'
    CapabilityType['WALLET'] = 'WALLET'
    CapabilityType['INTER_APP_AUDIO'] = 'INTER_APP_AUDIO'
    CapabilityType['MAPS'] = 'MAPS'
    CapabilityType['ASSOCIATED_DOMAINS'] = 'ASSOCIATED_DOMAINS'
    CapabilityType['PERSONAL_VPN'] = 'PERSONAL_VPN'
    CapabilityType['APP_GROUPS'] = 'APP_GROUPS'
    CapabilityType['HEALTHKIT'] = 'HEALTHKIT'
    CapabilityType['HOMEKIT'] = 'HOMEKIT'
    CapabilityType['WIRELESS_ACCESSORY_CONFIGURATION'] =
        'WIRELESS_ACCESSORY_CONFIGURATION'
    CapabilityType['APPLE_PAY'] = 'APPLE_PAY'
    CapabilityType['DATA_PROTECTION'] = 'DATA_PROTECTION'
    CapabilityType['SIRIKIT'] = 'SIRIKIT'
    CapabilityType['NETWORK_EXTENSIONS'] = 'NETWORK_EXTENSIONS'
    CapabilityType['MULTIPATH'] = 'MULTIPATH'
    CapabilityType['HOT_SPOT'] = 'HOT_SPOT'
    CapabilityType['NFC_TAG_READING'] = 'NFC_TAG_READING'
    CapabilityType['CLASSKIT'] = 'CLASSKIT'
    CapabilityType['AUTOFILL_CREDENTIAL_PROVIDER'] =
        'AUTOFILL_CREDENTIAL_PROVIDER'
    CapabilityType['ACCESS_WIFI_INFORMATION'] = 'ACCESS_WIFI_INFORMATION'
})((CapabilityType = exports.CapabilityType || (exports.CapabilityType = {})))
var CapabilityAllowedInstancesType
;(function(CapabilityAllowedInstancesType) {
    CapabilityAllowedInstancesType['ENTRY'] = 'ENTRY'
    CapabilityAllowedInstancesType['SINGLE'] = 'SINGLE'
    CapabilityAllowedInstancesType['MULTIPLE'] = 'MULTIPLE'
})(
    (CapabilityAllowedInstancesType =
        exports.CapabilityAllowedInstancesType ||
        (exports.CapabilityAllowedInstancesType = {}))
)
var CapabilitySettingKeyType
;(function(CapabilitySettingKeyType) {
    CapabilitySettingKeyType['ICLOUD_VERSION'] = 'ICLOUD_VERSION'
    CapabilitySettingKeyType['DATA_PROTECTION_PERMISSION_LEVEL'] =
        'DATA_PROTECTION_PERMISSION_LEVEL'
})(
    (CapabilitySettingKeyType =
        exports.CapabilitySettingKeyType ||
        (exports.CapabilitySettingKeyType = {}))
)
var CapabilityOptionKeyType
;(function(CapabilityOptionKeyType) {
    CapabilityOptionKeyType['XCODE_5'] = 'XCODE_5'
    CapabilityOptionKeyType['XCODE_6'] = 'XCODE_6'
    CapabilityOptionKeyType['COMPLETE_PROTECTION'] = 'COMPLETE_PROTECTION'
    CapabilityOptionKeyType['PROTECTED_UNLESS_OPEN'] = 'PROTECTED_UNLESS_OPEN'
    CapabilityOptionKeyType['PROTECTED_UNTIL_FIRST_USER_AUTH'] =
        'PROTECTED_UNTIL_FIRST_USER_AUTH'
})(
    (CapabilityOptionKeyType =
        exports.CapabilityOptionKeyType ||
        (exports.CapabilityOptionKeyType = {}))
)
